website/src/graphql/authExchange.ts

```diff
@@ -0,0 +1,70 @@
+import { authExchange } from "@urql/exchange-auth"
+import { CognitoIdToken } from "amazon-cognito-identity-js"
+import { makeOperation } from "urql"
+import { getCurrentUser, getIdToken } from "src/auth"
+import {
+  GRAPHQL_URL_READ,
+  GRAPHQL_URL_WRITE,
+  WP_GRAPHQL_URL,
+} from "src/graphql"
+
+const secondsSinceEpoch = () => Date.now() / 1000
+
+const isExpired = (token: CognitoIdToken) =>
+  token.getExpiration() <= secondsSinceEpoch()
+
+export const cognitoAuthExchange = () =>
+  authExchange<{ token: CognitoIdToken }>({
+    async getAuth({ authState }) {
+      if (!authState || isExpired(authState.token)) {
+        const token = await getIdToken()
+        if (token) return { token }
+      }
+
+      return null
+    },
+
+    willAuthError({ authState }) {
+      if (getCurrentUser()) {
+        // if we are signed in, we must have a valid token
+        return !authState || isExpired(authState.token)
+      } else if (authState) {
+        // we aren't signed in but we have a token, we need to get rid of it
+        return true
+      } else {
+        return false
+      }
+    },
```

Comment:
Let's make sure this is really correct

Reply:
I think something here isnt working as intended- I kept a window with a user logged in all weekend and the token did not expire.

Reply:
When you say that, do you mean the token was properly refreshed, or that we never detected the expiry and kept sending a bad token?